src/api/c/transform_coordinates.cpp

```diff
@@ -25,7 +25,15 @@ using namespace detail;
 template<typename T>
 Array<T> multiplyIndexed(const Array<T> &lhs, const Array<T> &rhs,
                          std::vector<af_seq> idx) {
-    return matmul(lhs, createSubArray(rhs, idx), AF_MAT_NONE, AF_MAT_NONE);
+    Array<T> rhs_subarr = createSubArray(rhs, idx);
+    int M    = lhs.dims()[0];
+    int N    = rhs_subarr.dims()[1];
+    dim_t d2 = std::max(lhs.dims()[2], rhs_subarr.dims()[2]);
+    dim_t d3 = std::max(lhs.dims()[3], rhs_subarr.dims()[3]);
+    const dim4 out_dims(M, N, d2, d3);
+    Array<T> out = createEmptyArray<T>(out_dims);
+    matmul(out, lhs, rhs_subarr, AF_MAT_NONE, AF_MAT_NONE);
+    return out;
```

Comment:
Using the latest change, you can even avoid calculating output dimensions here and let matmul do the allocation.